### PR TITLE
Refactors cult objectives to use datum team objectives, adds new cult objectives

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
refactor: Bloodcult objectives are now proper datum objectives
add: Blood cults now can have a new objective -- to cover the station in BLOOD!!
/:cl:

### CODE REVIEW

**Comment by MoreRobustThanYou** [comment]
@AnturK can you review this, you seem to know alot about antag datums n' shit

File: code/game/gamemodes/cult/cult_objectives.dm [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+/datum/objective/cult/eldergod/check_completion()
+	return (istype(GLOB.cult_narsie, /obj/singularity/narsie/large) || SSticker.mode.eldergod)
+
+/datum/objective/cult/spread_blood
+	success_mem_text = "The station has been properly prepared for the Geometer! <b>HAIL NAR-SIE!</b>"
+	var/spilltarget = 100
+
+/datum/objective/cult/spread_blood/New()
+	spilltarget = 100 + rand(0, GLOB.player_list.len * 3)
+	explanation_text = "We must prepare this place for the Geometer of Blood's coming. Spill blood over [spilltarget] floor tiles."
```
Comment (Xhuis) [inline code/game/gamemodes/cult/cult_objectives.dm]: Cult probably wins every round without trying tbh

**Comment by PKPenguin321** [comment]
Bad new objective

**Comment by ExcessiveUseOfCobblestone** [comment]
This pr is not atomized.

**Comment by MoreRobustThanYou** [comment]
@ExcessiveUseOfCobblestone I initially just wanted new objective, but I thought, might as well refactor too

**Comment by ExcessiveUseOfCobblestone** [comment]
REEEEEEEE

**Comment by ChangelingRain** [comment]
Atomize your pr so we can deny this terrible objective, please.

**Comment by MoreRobustThanYou** [comment]
I'm on a tablet, later.

File: code/datums/antagonists/cult.dm [hunk 2]
```diff
@@ -12,6 +12,10 @@
 	return ..()
 
 /datum/antagonist/cult/proc/add_objectives()
+	if(!SSticker.mode.cult_team)
```
Comment (AnturK) [inline code/datums/antagonists/cult.dm]: No, don't make it dependant on the mode. Initial assignment should create and set team along with objectives not the antag. Late conversions should use converter team. Also this has nothing to do with objectives so don't bundle it together. If you're really lazy make it static on the datum.

File: code/game/gamemodes/cult/ritual.dm [hunk 3]
```diff
@@ -197,9 +197,12 @@ This file contains the arcane tome files.
 	if(!src || QDELETED(src) || !Adjacent(user) || user.incapacitated() || !check_rune_turf(Turf, user))
 		return
 	if(ispath(rune_to_scribe, /obj/effect/rune/narsie))
-		if(!("eldergod" in SSticker.mode.cult_objectives))
+		if(!SSticker.mode.cult_team || !LAZYLEN(SSticker.mode.cult_team.objectives) || !(/datum/objective/cult/eldergod in SSticker.mode.cult_team.objectives))
```
Comment (AnturK) [inline code/game/gamemodes/cult/ritual.dm]: This should be a helper.

**Review by AnturK — CHANGES_REQUESTED** [review]

**Pull request closed without merging**